server: do not block shutdown with pending scram auth

ecwall opened the pull request:
Fixes #103108

Release note: None

ecwall committed:
Add TestServerArgs.AcceptSQLWithoutTLS

TestServerArgs.AcceptSQLWithoutTLS allows AcceptSQLWithoutTLS to be set instead
of Insecure to remove the need for TLS for tests that do not need it while
still enabling authentication.

Release note: None

ecwall committed:
Move server settings into serversettings package

Moving server settings into serversettings to prevent a circular
dependency between pgwire and server.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/103314)
<!-- Reviewable:end -->

ecwall committed:
Add TestServerInterface.Drain

This exposes the drain functionality to tests without addig a dependency
on the server package.

Release note: None

ecwall committed:
server: allow graceful node shutdown during pending scram auth

Fixes #103108

Release note (bug fix): Fixes bug that prevented nodes from shutting down
gracefully when a connection's SCRAM authentication was pending.

rafiss commented:
replaced by https://github.com/cockroachdb/cockroach/pull/104167
